src/minuteman_mesos_poller.erl

```diff
@@ -271,20 +271,28 @@ generate_ops(AgentIP, AgentVIPs, LashupVIPs) ->
     FlatAgentVIPs = flatten_vips(AgentVIPs),
     FlatLashupVIPs = flatten_vips(LashupVIPs),
     FlatVIPsToAdd = ordsets:subtract(FlatAgentVIPs, FlatLashupVIPs),
-    FlatLashupVIPsFromThisAgent = fetch_vips_from_agent(AgentIP),
+    FlatLashupVIPsFromThisAgent = fetch_vips_from_agent(AgentIP, FlatLashupVIPs),
```

Comment:
It seems like we're doing this in a very verbose way, and it might be more readable if we try an approach more like this:

```
ordsets:union(agent_vips(AgentIP, FlatLashupVIPs),
                        agent_vips(AgentIP, FlatLashupVIPs2)).

agent_vips(AgentIP, VipBEs) -> agent_vips(AgentIP, VipBEs, []).

agent_vips(_, [], Acc) -> Acc;
agent_vips(AIP, [#vip_be{backend_ip = AIP}=V|R], Acc) -> agent_vips(AIP, R, [V|Acc]);
agent_vips(AIP, [#vip_be2{agent_ip = AIP}=V2|R], Acc) -> agent_vips(AIP, R, [vipbe2_to_vipbe(V2)|Acc]);
agent_vips(AIP, [_|R], Acc) -> agent_vips(AIP, R, Acc).

vipbe2_to_vipbe(
  #vip_be2{protocol = Proto, vip_ip = VIP, vip_port = Port,
           backend_ip = BE, backend_port = BPort}) ->
    #vip_be{protocol = Proto, vip_ip = VIP, vip_port = Port,
            backend_ip = BE, backend_port = BPort}.
```

Thoughts?